From 1581e42efd34b5dd2b6ecade9807f28780775408 Mon Sep 17 00:00:00 2001
From: Paul Naoumenko <paulx.naoumenko@intel.com>
Date: Fri, 11 Oct 2013 11:39:12 +0200
Subject: [PORT FROM MAIN] wifi: increase ctrl_iface reply length.

BZ: 75654

Due to the new way Android Framework get the list of SSID,
It is needed to increase the reply length.
Otherwise, the list of SSID could be limited.
With this patch, list could contain about 100 APs.

Change-Id: I72599aaa8c1ed4375227865fd5cec6ce124bc754
Orig-Change-Id: I0024e770f545ed5f291f5d83c9f1b607c89ac557
Category: device enablement
Domain: CWS.WIFI-Common
Origin: upstream
Upstream-Candidate: no,proprietary
Signed-off-by: Jean Trivelly <jean.trivelly@intel.com>
---
 wpa_supplicant/ctrl_iface.c |    2 +-
 wpa_supplicant/wpa_cli.c    |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/wpa_supplicant/ctrl_iface.c b/wpa_supplicant/ctrl_iface.c
index b7f271d..39bbc8c 100644
--- a/wpa_supplicant/ctrl_iface.c
+++ b/wpa_supplicant/ctrl_iface.c
@@ -5246,7 +5246,7 @@ char * wpa_supplicant_ctrl_iface_process(struct wpa_supplicant *wpa_s,
 					 char *buf, size_t *resp_len)
 {
 	char *reply;
-	const int reply_size = 4096;
+	const int reply_size = 12288;
 	int ctrl_rsp = 0;
 	int reply_len;
 
diff --git a/wpa_supplicant/wpa_cli.c b/wpa_supplicant/wpa_cli.c
index a379d65..89996d7 100644
--- a/wpa_supplicant/wpa_cli.c
+++ b/wpa_supplicant/wpa_cli.c
@@ -398,7 +398,7 @@ static void wpa_cli_msg_cb(char *msg, size_t len)
 
 static int _wpa_ctrl_command(struct wpa_ctrl *ctrl, char *cmd, int print)
 {
-	char buf[4096];
+	char buf[12288];
 	size_t len;
 	int ret;
 
-- 
1.7.4.1

